components/scream/src/diagnostics/field_at_pressure_level.cpp

```diff
@@ -0,0 +1,84 @@
+#include "diagnostics/field_at_pressure_level.hpp"
+
+#include "ekat/std_meta/ekat_std_utils.hpp"
+#include "ekat/util/ekat_units.hpp"
+
+namespace scream
+{
+
+// =========================================================================================
+FieldAtPressureLevel::FieldAtPressureLevel (const ekat::Comm& comm, const ekat::ParameterList& params)
+  : AtmosphereDiagnostic(comm,params)
+  ,m_field_layout(m_params.get<FieldLayout>("Field Layout"))
+  ,m_field_units(m_params.get<ekat::units::Units>("Field Units"))
+  ,m_field_name(m_params.get<std::string>("Field Name"))
+  ,m_pressure_level(m_params.get<Real>("Field Target Pressure"))
+{
+  using namespace ShortFieldTagsNames;
+  EKAT_REQUIRE_MSG (ekat::contains(std::vector<FieldTag>{LEV,ILEV},m_field_layout.tags().back()),
+      "Error! FieldAtPressureLevel diagnostic expects a layout ending with 'LEV'/'ILEV' tag.\n"
+      " - field name  : " + m_field_name + "\n"
+      " - field layout: " + to_string(m_field_layout) + "\n");
+}
+
+// =========================================================================================
+void FieldAtPressureLevel::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
+{
+  using namespace ShortFieldTagsNames;
+  using namespace ekat::units;
+  const auto& gname  = m_params.get<std::string>("Grid Name");
+  auto m_grid = grids_manager->get_grid(gname);
+  const auto& grid_name = m_grid->name();
+  m_num_cols = m_grid->get_num_local_dofs();
+  auto num_levs = m_grid->get_num_vertical_levels();
+
+  constexpr int ps = Pack::n;
+
+  add_field<Required>(m_field_name, m_field_layout, m_field_units, gname);
+  if (m_field_layout.tags().back()==LEV) {
+    m_pres_name = "p_mid";
+    m_num_levs = num_levs;
+    FieldLayout pres_layout { {COL,LEV}, {m_num_cols,m_num_levs} };
```

Comment:
Isn't `pres_layout` simply `m_field_layout`? I think you could just do
```
m_pres_name = m_field_layout.tags().back()==LEV ? "p_mid" : "p_int";
add_field<Required>(m_pres_name,m_field_layout, Pa, gname, ps);
m_num_levs = m_field_layout.dims().back();
```

Reply:
Good point.